### DISCUSSION TITLE
How to apply external theme to a map?

### DISCUSSION THREAD

**Discussion by cdavila** [mapsforge/mapsforge#1477] [Q&A] [answered]
Hello all
I'm new to mapsforge. I'm trying to create a map with my own theme. Up to now, I've been able to create a test map with default theme by just running:
`osmosis --read-xml file="test.osm" --mapfile-writer file="test.map"`
According to [https://github.com/mapsforge/mapsforge/blob/master/docs/Rendertheme.md](url), in order to use my own theme, I should "activate `tileRendererLayer.setXmlRenderTheme(new ExternalRenderTheme(File))` at runtime", but I couldn't find how to do it. Could you give me some hint please?

**Comment by devemux86** [comment]
Please check the various samples, like the [MapsforgeMapViewer](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-samples-android/src/main/java/org/mapsforge/samples/android/MapsforgeMapViewer.java).

**Reply by cdavila** [reply]
I'm afraid I didn't explain well enough, sorry. What I want to know is if there's a way to build a map from OpenStreetMap data using osmosis + map-writer.jar (from Sonatype repository) + custom theme. If not, how should I integrate my own theme into map-writer?

**Comment by devemux86** [comment, marked as answer]
Mapsforge map writer is used to create maps, please see the [map creation guide](https://github.com/mapsforge/mapsforge/blob/master/docs/MapCreation.md).

Themes are created separately and loaded inside the apps, please see the samples.
